source/discovery/carla-discovery.cpp

```diff
@@ -1122,6 +1123,34 @@ static void do_vst_check(lib_t& libHandle, const char* const filename, const boo
         else
             cVendor.clear();
 
+        // get category
+        const intptr_t effCategory = effect->dispatcher(effect, effGetPlugCategory, 0, 0, NULL, 0.0f);
+        category = PLUGIN_CATEGORY_NONE;
```

Comment:
this should not be here, as you are setting the variable again right after in the switch case.
please add a default case in the switch, and set this there instead.